consensus/core/src/core.rs

```diff
@@ -732,6 +821,18 @@ impl Core {
                 .try_unsuspend_blocks_for_latest_gc_round();
 
             committed_sub_dags.extend(subdags);
+
+            fail_point!("consensus-after-handle-commit");
+        }
+
+        // Sanity check: for commits that have been linearized using the certified commits, ensure that the same sub dag has been committed.
+        for sub_dag in &committed_sub_dags {
+            if let Some(commit_ref) = certified_commits_map.remove(&sub_dag.commit_ref.index) {
```

Comment:
When can `remove()` return `None`?

Reply:
The commit rule here runs within a loop. The order of processing is the following:
1. Try to commit all the certified leaders - if any exist - on the `certified_commits` vector. Do this until this vector is empty
2. Once `certified_commits` is empty then try to run the commit rule `try_decide`.
3. If there is any leader schedule change, in the meantime, will commit up to leader schedule and then continue on the next loop

Now, give the hybrid mode operation it's possible that during the loop we indeed commit some certified leaders - which should exist in the `certified_commits_map` , and some which aren't (via the common commit rule). For the later case we don't expect them to be found in the `certified_commits_map`.